common/src/stack/command/stack/commands/list/attr/plugin_global_intrinsic.py

```diff
@@ -0,0 +1,81 @@
+# @copyright@
+# Copyright (c) 2006 - 2019 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+
+import stack.commands
+import stack.mq
+from stack.util import flatten
+
+from ipaddress import IPv4Network
+
+
+class Plugin(stack.commands.Plugin):
+	"""
+	Output intrinstic attrs for global scope
+	"""
+
+	def provides(self):
+		return 'global_intrinsic'
+
+	def run(self, scope_mappings):
+		# Figure out the output targets
+		node_ids = []
+		targets = []
+		for scope_mapping in scope_mappings:
+			# We only need to run for host or global scope
+			if scope_mapping.scope == 'host':
+				node_ids.append(scope_mapping.node_id)
+			elif scope_mapping.scope == 'global':
+				targets.append('')
+			else:
+				continue
+
+		if node_ids:
+			targets.extend(flatten(self.db.select(
+				'nodes.name FROM nodes WHERE nodes.id IN %s',
+				(node_ids,)
+			)))
+
+		# Get various kickstart networking data
+		output_rows = []
+		if targets:
+			for ip, hostname, zone, address, netmask, network_name in self.db.select("""
+				networks.ip, IF(networks.name IS NOT NULL, networks.name, nodes.name),
+				subnets.zone, subnets.address, subnets.mask, subnets.name
+				FROM networks
+				INNER JOIN subnets ON subnets.id = networks.subnet
+				INNER JOIN nodes ON nodes.id = networks.node
+				INNER JOIN appliances ON appliances.id = nodes.appliance
+				WHERE appliances.name = 'frontend'
+				AND (subnets.name = 'public' OR subnets.name = 'private')
+			"""):
+				network = IPv4Network(f"{address}/{netmask}")
+
+				if network_name == 'private':
+					for target in targets:
+						output_rows.append([target, 'global', 'const', 'Kickstart_PrivateKickstartHost', ip])
+						output_rows.append([target, 'global', 'const', 'Kickstart_PrivateAddress', ip])
+						output_rows.append([target, 'global', 'const', 'Kickstart_PrivateHostname', hostname])
+						output_rows.append([target, 'global', 'const', 'Kickstart_PrivateBroadcast', str(network.broadcast_address)])
+						output_rows.append([target, 'global', 'const', 'Kickstart_PrivateDNSDomain', zone])
+						output_rows.append([target, 'global', 'const', 'Kickstart_PrivateNetwork', address])
+						output_rows.append([target, 'global', 'const', 'Kickstart_PrivateNetmask', netmask])
+						output_rows.append([target, 'global', 'const', 'Kickstart_PrivateNetmaskCIDR', str(network.prefixlen)])
```

Comment:
Can we kill some of these?

We can calculate broadcast if we have the network and netmask.  Likewise, we need the netmask or the CIDR, but not both.  Is there any point in keeping PrivateKickstartHost *and* PrivateAddress, given that they're already the same?

Same with the `Kickstart_Public*`.

Reply:
I got no problems killing some of them. I just copied the ones that the old code was creating. I have no clue what ones are safe to remove, but you seem to. Maybe that should be a separate feature branch after this code gets merged, in case something we kill turned out to be needed?

Reply:
Yeah, that's fair.